File: frontend/src/types/weather.ts
import type { DataQuality, SourceItem } from './api';
import type { AreaSummary } from './area';

export type WeatherPeriod = {
  start_time: string;
  end_time: string;
  temperature: number;
  temperature_unit: string;
  precipitation_probability: number | null;
  wind_speed: string;
  wind_direction: string;
  short_forecast: string;
  detailed_forecast?: string;
  is_daytime: boolean;
};

export type WeatherResponse = {
  area: AreaSummary;
  weather: {
    mode: 'hourly_summary' | 'target_time';
    target_time: string | null;
    periods: WeatherPeriod[];
  };
  data_quality: DataQuality;
  source: SourceItem[];
  updated_at: string;
  expires_at: string | null;
};

File: backend/api-gateway/app/models/weather.py
from pydantic import BaseModel
from app.models.area import AreaSummary
from app.models.common import DataQuality, SourceItem, WeatherPeriod


class WeatherPayload(BaseModel):
    mode: str
    target_time: str | None = None
    periods: list[WeatherPeriod]


class WeatherResponse(BaseModel):
    area: AreaSummary
    weather: WeatherPayload
    data_quality: DataQuality
    source: list[SourceItem]
    updated_at: str
    expires_at: str | None = None
